src/Pact/Types/Term.hs

```diff
@@ -1253,7 +1266,41 @@ termEq (TSchema a b c d _) (TSchema e f g h _) = a == e && b == f && c == g && a
 termEq (TVar a _) (TVar b _) = a == b
 termEq _ _ = False
 
+instance Eq1 Guard where liftEq = $(makeLiftEq ''Guard)
+instance Eq1 UserGuard where liftEq = $(makeLiftEq ''UserGuard)
+instance Eq1 Term where liftEq = $(makeLiftEq ''Term)
+instance Eq1 BindPair where liftEq = $(makeLiftEq ''BindPair)
+instance Eq1 App where liftEq = $(makeLiftEq ''App)
+instance Eq1 BindType where liftEq = $(makeLiftEq ''BindType)
+instance Eq1 ConstVal where liftEq = $(makeLiftEq ''ConstVal)
+instance Eq1 DefcapMeta where liftEq = $(makeLiftEq ''DefcapMeta)
+instance Eq1 DefMeta where liftEq = $(makeLiftEq ''DefMeta)
+instance Eq1 Def where liftEq = $(makeLiftEq ''Def)
+instance Eq1 ModuleDef where liftEq = $(makeLiftEq ''ModuleDef)
+instance Eq1 Module where liftEq = $(makeLiftEq ''Module)
+instance Eq1 Governance where liftEq = $(makeLiftEq ''Governance)
+instance Eq1 ObjectMap where liftEq = $(makeLiftEq ''ObjectMap)
+instance Eq1 Object where liftEq = $(makeLiftEq ''Object)
+instance Eq1 Step where liftEq = $(makeLiftEq ''Step)
+
+instance Show1 Guard where liftShowsPrec = $(makeLiftShowsPrec ''Guard)
+instance Show1 UserGuard where liftShowsPrec = $(makeLiftShowsPrec ''UserGuard)
+instance Show1 Term where liftShowsPrec = $(makeLiftShowsPrec ''Term)
+instance Show1 BindPair where liftShowsPrec = $(makeLiftShowsPrec ''BindPair)
+instance Show1 App where liftShowsPrec = $(makeLiftShowsPrec ''App)
+instance Show1 ObjectMap where liftShowsPrec = $(makeLiftShowsPrec ''ObjectMap)
+instance Show1 Object where liftShowsPrec = $(makeLiftShowsPrec ''Object)
+instance Show1 BindType where liftShowsPrec = $(makeLiftShowsPrec ''BindType)
+instance Show1 ConstVal where liftShowsPrec = $(makeLiftShowsPrec ''ConstVal)
+instance Show1 DefcapMeta where liftShowsPrec = $(makeLiftShowsPrec ''DefcapMeta)
+instance Show1 DefMeta where liftShowsPrec = $(makeLiftShowsPrec ''DefMeta)
+instance Show1 Def where liftShowsPrec = $(makeLiftShowsPrec ''Def)
+instance Show1 ModuleDef where liftShowsPrec = $(makeLiftShowsPrec ''ModuleDef)
+instance Show1 Module where liftShowsPrec = $(makeLiftShowsPrec ''Module)
+instance Show1 Governance where liftShowsPrec = $(makeLiftShowsPrec ''Governance)
+instance Show1 Step where liftShowsPrec = $(makeLiftShowsPrec ''Step)
```

Comment:
Why get rid of the `derive` macros? Deprecated?

Reply:
Same reason as above. Unentangling TH related dependencies. By moving the TH splice into the body of the instance declaration, the declaration itself comes into scope before it is used in another splice.